portaudio.go

```diff
@@ -731,7 +756,13 @@ func streamCallback(inputBuffer, outputBuffer unsafe.Pointer, frames C.ulong, ti
 		}
 	}()
 
-	s := (*Stream)(userData)
+	id := uintptr(userData)
+	mu.Lock()
```

Comment:
`mu` could be a `sync.RWMutex` and this block could call `mu.RLock`. This would prevent multiple uses of streamCallback locking over the same resource during reads.